src/python/pants/source/source_root.py

```diff
@@ -93,6 +100,12 @@ def find_by_path(self, path):
       # TODO: Remove this logic. It should be an error to have no matching source root.
       return SourceRoot(path, [], SourceRootCategories.UNKNOWN)
 
+  def traverse(self) -> Set[str]:
+    return self._trie.traverse()
+
+  def trie_find(self, dir_path: str) -> SourceRoot:
+    return self._trie.find(dir_path)
+
   def all_roots(self):
```

Comment:
Do we kill this method now?

Reply:
No it's still being used in once place (cf. https://github.com/pantsbuild/pants/pull/8199/commits/29713c06f07a843973cd41fb303807a5d40e7341 )

Reply:
Fair enough.